apps/aehttp/test/aehttp_ws_test_utils.erl

```diff
@@ -0,0 +1,155 @@
+-module(aehttp_ws_test_utils).
+-behaviour(websocket_client).
+
+%% API
+-export([start/2,
+         send/4,
+         subscribe_for_event/3,
+         unsubscribe_for_event/3,
+         wait_for_event/2,
+         wait_for_event/3,
+         stop/1]).
+
+%% behaviour exports
+-export([
+         init/1,
+         onconnect/2,
+         ondisconnect/2,
+         websocket_handle/3,
+         websocket_info/3,
+         websocket_terminate/3
+        ]).
+
+start(Host, Port) ->
+    crypto:start(),
+    ssl:start(),
+    WsAddress = "ws://" ++ Host ++ ":" ++ integer_to_list(Port) ++ "/websocket",
+    ct:log("connecting to ~p", [WsAddress]),
+    {ok, Pid} = websocket_client:start_link(WsAddress, ?MODULE, self()),
+    ok = wait_for_event(websocket, connected),
```

Comment:
I see - on init argument (self()) is automatically subscribed to waiting_connected, and onconnect subscription is removed and message is sent to caller. This may lead to unexpected message in mailbox of caller of aehttp_ws_test_utils:start (in case of timeout/slowness/race). It would be nice to have a line of comment documenting this potential message in.